otx/cli/tools/train.py

```diff
@@ -144,15 +135,62 @@ def parse_args():
         help="Total number of workers in a worker group.",
     )
 
-    add_hyper_parameters_sub_parser(parser, hyper_parameters)
-
-    return parser.parse_args(), template, hyper_parameters
+    return parser
 
 
-def main():  # pylint: disable=too-many-branches
+# pylint: disable=too-many-branches, too-many-statements
+def main():
     """Main function that is used for model training."""
-    # Dynamically create an argument parser based on override parameters.
-    args, template, hyper_parameters = parse_args()
+    parser = get_parser()
+
+    args, _ = parser.parse_known_args()
+    train_workspace_path = os.getcwd()
+
+    default_workspace_components = {"tempate_path": "./template.yaml", "data_path": "./data.yaml"}
+    has_template_yaml = Path(default_workspace_components["template_path"]).exists()
+    has_data_yaml = Path(default_workspace_components["data_path"]).exists()
+
+    # Possible scenario 1: otx build --> otx train
+    # Possible scenario 2: otx build --> otx train --template=${TEMPLATE}
+    # Possible scenario 3: otx train --train-data-roots=${PATH}
+    # So, we need to check whethere there is workspace or not
+    use_workspace = has_template_yaml and has_data_yaml
+    if not use_workspace:
+        # Prepare build
+        train_workspace_path = args.save_model_to
+
+        # If an user gives a weird path, then automatically selects default template by using build function.
+        if Path(args.template).exists():
+            template = find_and_parse_model_template(args.template)
+        else:
+            print(f"Can't find {args.template}, the default template will be used to train. ")
+            template = None
+        # Build
+        builder = Builder()
+        build(
+            builder=builder,
+            train_data_roots=args.train_data_roots,
+            val_data_roots=args.val_data_roots,
+            workspace_root=train_workspace_path,
+            template=template,
+        )
```

Comment:
I'm not sure but, doesn't `template` variable have to re-initialized as newly build template.yaml?

Reply:
Could you explain more detail? 

If you mean, `template` variable don't have to re-initialized in `build` function ?

If a user doesn't use workspace, then we need to build first. Otherwise, if the user use workspace, then we can use the `template.yaml` located in workspace.

Reply:
I mean as below.
```
        template = find_and_parse_model_template(args.template)
        builder = Builder()
        build(
            builder=builder,
            train_data_roots=args.train_data_roots,
            val_data_roots=args.val_data_roots,
            workspace_root=train_workspace_path,
            template=template,
        )
        template = find_and_parse_model_template(os.path.join(train_workspace_path, "template.yaml") <= not necessary?
```

Reply:
Good point, I missed. I'll edit

Reply:
Could you check this https://github.com/openvinotoolkit/training_extensions/pull/1592/commits/6d763e1826429a1b41760fab70550fe47e63f245?

Reply:
Is template file made in current directory after build? then if args.template doesn't exists and isn't default path (i.e. not ./template.yaml), then args.template is loaded even if newly made template path is "./template.yaml"

Reply:
If user gives an correct path for `args.template`, then we need to use this argument for training.

So, I think #L180 is necessary. If a user gives a not existed template, then the user must meet the error. Because OTX can't training with not-existed template

Reply:
I think if `args.template = default_workspace_components["template_path"]` right after build, then user don't have to meet the error. Also, user get f"Can't find {args.template}, the default template will be used to train." message before build. So user may think training works well without any error, I think. How do you think about?

Reply:
> I think if `args.template = default_workspace_components["template_path"]` right after build, then user don't have to meet the error.

Actually, we can't avoid the error when a user gives a weird template path. And in this case, making an error is more natural thing instead of selecting default template.

If a user gives `otx train --template=${NOT EXISTED TEMPLATE} --train-data-roots=${PATH}` in the workspace, user also meet the error. 

> Also, user get f"Can't find {args.template}, the default template will be used to train." message before build.

Even in the current implementation, the user receives this message before `build()`.

Reply:
As far as I understand, User can execute train command without error even if passing wrong template argument.
Below is how I understand code flow. I know I may misunderstand. If then, I'll really appreciate if you let me know that :)
```
    ...
    has_template_yaml = Path(default_workspace_components["template_path"]).exists() => doesn't exists, so it's False
    has_data_yaml = Path(default_workspace_components["data_path"]).exists()

    use_workspace = has_template_yaml and has_data_yaml
    if not use_workspace:
        # Prepare build
        train_workspace_path = args.save_model_to

        # If an user gives a weird path, then automatically selects default template by using build function.
        if Path(args.template).exists():
            template = find_and_parse_model_template(args.template)
        else: => because it doesn't exist, this branch is executed. So, template is initialized as None
            print(f"Can't find {args.template}, the default template will be used to train. ")
            template = None
        # Build
        builder = Builder()
        build(
            builder=builder,
            train_data_roots=args.train_data_roots,
            val_data_roots=args.val_data_roots,
            workspace_root=train_workspace_path,
            template=template, => build is executed with None template.
        )

    # Update configurations made by builder
    # When an user gives template argument, need to overwrite it.
    if args.template != default_workspace_components["template_path"]: => even if template is made automatically in build, this branch is executed and raise error.
        template = find_and_parse_model_template(args.template)
    else:
        template = find_and_parse_model_template(os.path.join(train_workspace_path, "template.yaml"))
    ...
```